docs/naps/8-telemetry.md

```diff
@@ -0,0 +1,190 @@
+ 
+ ## Motivation and Scope
+
+With the growth of napari, the standard feedback loop through napari community meetings and napari-related events at conferences has reached its capacity. Also, we collect many feature requests for which we cannot find volunteers for implementation. 
+
+To have the possibility of sustainable development of the project we will either need to rely on paid contractors or on companies donating employee time managed by the core devs.
+
+Both scenarios require us to provide some information about the estimated number of users to prove to potential funders that their donation/grant will be used in a valuable way. 
+
+Adding the option for monitoring plugin usage allows us to identify the most important plugins and try to establish cooperation with their maintainers to reduce the probability that the plugin will not be ready for a new napari release. Such monitoring could contain not only the list of installed plugins but also which commands and contributions are used most often.
+
+Also collecting information about data types and their size will provide valuable information about the typical use cases of napari.
+
+Still, users need to be able to opt out of such monitoring, and adjust the level of detail of the information that is sent to the napari server.
+ 
+  
+ ## Detailed Description
+
+`napari-telemetry` will be a package responsible for collecting and sending telemetry data to the napari server. It will be installed after user confirmation. It will contain callbacks for data collection, and utils for storage and sending. Also, this package will contain utils for validating if the user has agreed to telemetry. 
+
+In the main package, there is a need to add code to ask users if they want to enable telemetry. This code should be executed only once per environment.
+
+Telemetry should contain following ways to disable it:
+
+1. Disable in settings
+2. uninstall `napari-telemetry` package
+3. Environment variable `NAPARI_TELEMETRY=0`
+4. Full list of endpoints used for collecting telemetry, that could be filtered on the firewall level.
+
+The user should be able to adjust the telemetry level of detail. The following levels are proposed:
+
+1. `none` - no telemetry is collected
+2. `basic` - information about the napari version, python version, OS, and CPU architecture is collected and if it is the first report by the user. There is also a user identifier created based on computer details that will be rerendered each week to prevent tracking the user, but allow to not count a user multiple times. 
+3. `middle` - same as in `basic` but also information about the list of installed plugins and their versions is collected. We take care to not collect data about plugins that are not intended to be public, so we will not collect information about plugins searchable as napri plugin using plugin dialog or napri-hub. We also will not collect information about plugins that are installed in non stable version.
```

Comment:
I believe `basic` and `middle` could be combined into a single level. From the dev's perspective, `basic` does not (arguably) bring much actionable information. From the user's perspective, these two levels are likely to be percieved as providing similar levels of anonymity.

Reply:
In theory middle allow to identification of user base on a unique set of plugins. Where base still allows to have basic knowledge about os, new python adaptation etc.

Reply:
I agree @Czaki I would like to keep the distinction there.